UnityProject/Assets/Scripts/Items/Tools/Spray.cs

```diff
@@ -0,0 +1,178 @@
+                spray2Sprite.Rotate(new Vector3(0, 0, 90 * -yDir));
+                spray3Sprite.Rotate(new Vector3(0, 0, 90 * -yDir));
+            }
+            //Cleaning spray
+            if (effect == "cleaning")
+            {
+                for (int j = 0; j < 3; j++)
+                {
+                    //Clean and move if no wall
+                    CleanTile(position + new Vector3(j * xDir, j * yDir));
+                    if (!(MatrixManager.GetMetaDataAt(SprayGO.transform.position.CutToInt() + new Vector3(xDir, yDir).CutToInt()).IsOccupied))
+                    {
+                        SprayGO.transform.Translate(new Vector3(xDir, yDir));
+                    }
+                    yield return new WaitForSecondsRealtime(0.5f);
+                }
+            }
+            //Extinguishing spray
+            else if (effect == "extinguish")
+            {
+                SprayGO2.SetActive(true);
+                SprayGO3.SetActive(true);
+                for (int j = 0; j < 3; j++)
+                {
+                    //Put out fire and move if no walls
+                    ExtinguishTile(position + new Vector3(j * xDir, j * yDir));
+                    ExtinguishTile(position + new Vector3((j * xDir) - yDir, (j * yDir) + xDir));
+                    ExtinguishTile(position + new Vector3((j * xDir) + yDir, (j * yDir) - xDir));
+                    if(!(MatrixManager.GetMetaDataAt(SprayGO.transform.position.CutToInt() + new Vector3(xDir, yDir).CutToInt()).IsOccupied))
+                    {
+                        SprayGO.transform.Translate(new Vector3(xDir, yDir));
+                    }
+                    if (!(MatrixManager.GetMetaDataAt(SprayGO2.transform.position.CutToInt() + new Vector3(xDir, yDir).CutToInt()).IsOccupied))
+                    {
+                        SprayGO2.transform.Translate(new Vector3(xDir, yDir));
+                    }
+                    if (!(MatrixManager.GetMetaDataAt(SprayGO3.transform.position.CutToInt() + new Vector3(xDir, yDir).CutToInt()).IsOccupied))
+                    {
+                        SprayGO3.transform.Translate(new Vector3(xDir, yDir));
+                    }
+                    yield return new WaitForSecondsRealtime(0.5f);
+                }
+            }
+            else
+            {
+                for (int j = 0; j < 3; j++)
+                {
+                    //Move if no walls
+                    if (!(MatrixManager.GetMetaDataAt(SprayGO.transform.position.CutToInt() + new Vector3(xDir, yDir).CutToInt()).IsOccupied))
+                    {
+                        SprayGO.transform.Translate(new Vector3(xDir, yDir));
+                    }
+                    SprayGO.transform.Translate(new Vector3(xDir, yDir));
+                    yield return new WaitForSecondsRealtime(0.5f);
+                }
+            }
+            //Destroy spray objects after 3 moves
+            GameObject.Destroy(SprayGO);
```

Comment:
You used PoolManager to create the objects, which is good! But if you use GameObject.Destroy, they won't go back into the object pool, negating the benefit of object pooling. Just use PoolManager.PoolNetworkDestroy and it should be fine.